packages/core/tangleold/ratesetter.go

```diff
@@ -40,17 +71,45 @@ var (
 
 // RateSetterParams represents the parameters for RateSetter.
 type RateSetterParams struct {
-	Enabled          bool
+	Mode             RateSetterModeType
 	Initial          float64
 	RateSettingPause time.Duration
 }
 
 // endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-// region RateSetter ///////////////////////////////////////////////////////////////////////////////////////////////////
+// region RateSetter interface ///////////////////////////////////////////////////////////////////////////////////////////////////
+
+type RateSetter interface {
+	// setup and shutdown
+	Setup()
+	Shutdown()
+	// web API
+	Rate() float64
+	Estimate() time.Duration
+	Size() int
+	// getter for events
+	RateSetterEvents() *RateSetterEvents
```

Comment:
Just call it `Events`?

Reply:
Can't have field and method with same name

Reply:
nvm, I see next suggestion to make the field private